eldritch/test_events.py

```diff
@@ -262,6 +296,151 @@ def testMoveMultipleSpaces(self):
+
+    self.advance_turn(self.state.turn_number, "movement")
+    movement = self.resolve_to_choice(CityMovement)
+    self.assertFalse(movement.is_resolved())
+    self.assertEqual(self.char.movement_points, 4)
+    self.assertNotIn("Graveyard", movement.choices)
+    self.assertIn("Rivertown", movement.choices)
+    movement.resolve(self.state, "Rivertown")
+
+    movement = self.resolve_to_choice(CityMovement)
+    self.assertIn("Graveyard", movement.choices)
+    movement.resolve(self.state, "Graveyard")
+
+    choice = self.resolve_to_choice(MultipleChoice)
+
+    self.assertEqual(choice.choices, ["Fight", "Evade"])
+    choice.resolve(self.state, "Fight")
+
+    next_choice = self.resolve_to_choice(MultipleChoice)
+    self.assertEqual(sorted(next_choice.choices), ["Fight", "Flee"])
+    next_choice.resolve(self.state, "Fight")
+
+    third_choice = self.resolve_to_choice(CombatChoice)
+    self.assertFalse(third_choice.choices)
+    third_choice.resolve(self.state, [])
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=5)):
+      movement = self.resolve_to_choice(CityMovement)
+
+    self.assertFalse(movement.choices)
+    movement.resolve(self.state, movement.none_choice)
+    self.resolve_until_done()
+
+    self.assertTrue(movement.is_resolved())
+    self.assertEqual(self.char.place.name, "Rivertown")
+    self.assertEqual(self.char.movement_points, 0)
+    # self.assertIn(cultist, self.char.possessions)
+    # TODO: take the monster as a trophy
+
+  def testMoveMultipleThroughMonsterFailedEvade(self):
+    zombie = monsters.Zombie()
+    self.state.monsters.append(zombie)
+    zombie.place = self.state.places["Rivertown"]
+    self.char.place = self.state.places["Downtown"]
+
+    self.advance_turn(self.state.turn_number, "movement")
+    movement = self.resolve_to_choice(CityMovement)
+    self.assertFalse(movement.is_resolved())
+    self.assertEqual(self.char.movement_points, 4)
+    self.assertNotIn("Graveyard", movement.choices)
+    self.assertIn("Rivertown", movement.choices)
+    movement.resolve(self.state, "Rivertown")
+
+    movement = self.resolve_to_choice(CityMovement)
+    self.assertIn("Graveyard", movement.choices)
+    movement.resolve(self.state, "Graveyard")
+
+    choice = self.resolve_to_choice(MultipleChoice)
+
+    self.assertEqual(choice.choices, ["Fight", "Evade"])
+    choice.resolve(self.state, "Evade")
+
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=3)):
+      next_choice = self.resolve_to_choice(MultipleChoice)
+    self.assertEqual(sorted(next_choice.choices), ["Fight", "Flee"])
+    next_choice.resolve(self.state, "Flee")
+
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=5)):
+      movement = self.resolve_to_choice(CityMovement)
+
+    self.assertFalse(movement.choices)
+    movement.resolve(self.state, movement.none_choice)
+    self.resolve_until_done()
```

Comment:
You should add a comment here - using resolve_until_done() also tests that you don't have to evade the zombie again when you're done with your movement.